### PR TITLE
Topic/tl ucp ep close sync

### PR DESCRIPTION
**Pull request by vspetrov**
## What
If OOB is provided for UCC context then we should launch a barrier after endpoints have been closed. Otherwise, we have to create ucp ep endpoints with ERR_HANDLER (to avoid segv during ep_flush). This may results in worse perf do to suboptimal ucp ep lanes selection.

## Why ?
W/o the fix segv is likely to happen in ucc_tl_ucp_context_destroy

### CODE REVIEW

**Comment by vspetrov** [comment]
@Sergei-Lebedev @manjugv @bureddy plz review. made according to our today's discussion.

File: src/components/tl/ucp/tl_ucp_ep.c [hunk 1]
```diff
@@ -17,6 +23,13 @@ static inline ucc_status_t ucc_tl_ucp_connect_ep(ucc_tl_ucp_context_t *ctx,
     ep_params.field_mask = UCP_EP_PARAM_FIELD_REMOTE_ADDRESS;
     ep_params.address    = (ucp_address_t*)address->addr;
 
+    if (!UCC_TL_CTX_HAS_OOB(ctx)) {
+        ep_params.err_mode        = UCP_ERR_HANDLING_MODE_PEER;
+        ep_params.err_handler.cb  = ucc_tl_ucp_err_handler;
+        ep_params.err_handler.arg = NULL;
+        ep_params.field_mask     |= UCP_EP_PARAM_FIELD_ERR_HANDLING_MODE |
+                                    UCP_EP_PARAM_FIELD_ERR_HANDLER;
+    }
```
Comment (manjugv) [inline src/components/tl/ucp/tl_ucp_ep.c:32]: @vspetrov Should we warn the user about non-optimal behavior?

**Review by manjugv — APPROVED** [review]
Minor comment. Otherwise, it looks fine.

**Review by Sergei-Lebedev — APPROVED** [review]

File: src/components/tl/ucp/tl_ucp_ep.c (hunk 1 above)
Comment (vspetrov) [reply src/components/tl/ucp/tl_ucp_ep.c:32]: @manjugv i think not, we might add some note in the manual or readme

**Pull request merged**